### PR TITLE
Partial #6914 job_agent separated terminating vs destroying

### PR DESCRIPTION
**Pull request by robnagler**
- does not cancel the sbatch job when terminating
- job_agent _SBATCH_ID_FILE write
- job_supervisor concept of verify_status but doesn't change semantics yet
- pkcli.elegant-schema better approach to updating schema
- const.DEV_SRC_RADIASOFT_DIR

### CODE REVIEW

**Review by e-carlin — CHANGES_REQUESTED** [review]
Some style comments. Not sure what behavior should be implemented with this PR but I saw this:
- start an sbatch job
- kill -9 supervisor # No chance to terminate so agent is left running
- restart supervisor
- login to sbatch agent
- simulation is canceled

Is this sim supposed to be canceled?

**Comment by robnagler** [comment]
> * start an sbatch job
> * kill -9 supervisor # No chance to terminate so agent is left running
> * restart supervisor
> * login to sbatch agent
> * simulation is canceled
> 
> Is this sim supposed to be canceled?

This is caused by the scancel in sbatch.py when the agent starts. This PR didn't fix that.

File: sirepo/pkcli/job_agent.py [hunk 1]
```diff
@@ -45,10 +46,16 @@
 
 _PID_FILE = "job_agent.pid"
 
-_PY2_CODES = frozenset(())
+_SBATCH_ID_FILE = "sirepo_sbatch_id"
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/pkcli/job_agent.py:49]: Yes, that was the idea, but the follow on PR will do things differently. :)

**Review by e-carlin — APPROVED** [review]
LGTM then. Do what you will with my style comments.

File: sirepo/pkcli/job_agent.py [hunk 2]
```diff
@@ -686,12 +697,14 @@ async def _await_start_ready(self):
...
-    def destroy(self):
+    def destroy(self, terminating=False):
+        self._destroying = True
+        self._terminating = terminating
         if self._status_cb:
             self._status_cb.stop()
             self._status_cb = None
         self._start_ready.set()
-        if self._sbatch_id:
+        if self._sbatch_id and not self._terminating:
```
[the remark this answers is not in the document]
Comment (robnagler) [reply sirepo/pkcli/job_agent.py:707]: Kind of depends on which place you are reading. Inside terminate, terminating is set to True and that cascades down several levels. The intent in terminate() is to signal that the agent is terminating. This use parallels `self._destroying`, which indicates the `_Cmd` is in destroy(). There are many conditionals which use `_destroying` for different purposes, but they are all relying on the fact that the object is in destroy(). 

In this particular case, there's only one reason to know that the process is in terminate(), but there could be other reasons, and the way the value cascades in the call to terminate() makes sense (to me at least). It's also a private attribute of the class so it's use is localized, e.g. the Dispatcher doesn't refer to it.

That's the reasoning of "what the heck is going on". It's just a different perspective: from the whole process vs the specific conditional.

**Pull request merged**